LibGit2Sharp/Branch.cs

```diff
@@ -226,7 +236,17 @@ private Branch ResolveTrackedBranch()
                 return null;
             }
 
-            string trackedReferenceName = Proxy.git_branch_upstream_name(repo.Handle, CanonicalName);
+            string trackedReferenceName = null;
+            try
+            {
+                trackedReferenceName = Proxy.git_branch_upstream_name(repo.Handle, CanonicalName);
+            }
+            catch (InvalidSpecificationException)
+            {
+                // we have no way to pro-actively check this
```

Comment:
@joshvera the shitty part about this scenario is that it totally nukes the graph in this scenario, so disabling tracking seems like a decent tradeoff. Also, the work required to "catch up" to libgit2/sharp is non-trivial at the moment. :cry: